tasks/main.yml

```
@@ -122,6 +122,53 @@
   tags:
     - wg-config
 
+- name: Anticipate configuration changes
+  ansible.builtin.template:
+    src: etc/wireguard/wg.conf.j2
+    dest: "{{ wireguard_remote_directory }}/{{ wireguard_interface }}.conf"
+    owner: "{{ wireguard_conf_owner }}"
+    group: "{{ wireguard_conf_group }}"
+    mode: "{{ wireguard_conf_mode }}"
+  check_mode: yes
+  diff: yes
+  changed_when: no
+  register: wg_config
```

Internal variables start with `wireguard__`:
```suggestion
  register: wireguard__register_configuration
```